### PR TITLE
Deduplicate Switch/If expressions according to the intended use case

### LINKED ISSUE #91: Question: why are IfThen and SwitchExpression messages near exact copies of each other?
What is the use case for having `IfThen` and `SwitchExpression` messages that are effectively exact copies of each other?

```protobuf
  message IfThen {
    repeated IfClause ifs = 1;
    Expression else = 2;

    message IfClause {
      Expression if = 1;
      Expression then = 2;
    }
  }

  message SwitchExpression {
    repeated IfValue ifs = 1;
    Expression else = 2;

    message IfValue {
      Expression if = 1;
      Expression then = 2;
    }
  }
```

I understand the difference between the two different types of `CASE` expressions in SQL.

Can't we represent `SwitchExpression` in terms of an `IfThen` with an equality condition?

If we still need to distinguish between the two ... cases (pun intended), can we refactor this to use an `enum` for differentiating instead of two effectively identical constructs?

### PR DESCRIPTION
**Pull request by cpcloud**
Closes #91

### CODE REVIEW

File: binary/expression.proto [hunk 1]
```diff
@@ -148,32 +148,31 @@ message Expression {
     }
   }
 
-  message IfThen {
-
-    repeated IfClause ifs = 1;
+  message CasePredicateExpression {
```
Comment (jacques-n) [inline binary/expression.proto:151]: Can we keep the IfThen names as opposed to changing this? I think it does a better job of clarifying distinction from the other type. It also avoids being overly sql centric for a pretty standard programming concept.

Comment (cpcloud) [reply binary/expression.proto:151]: That's fair. I can keep the original names.

Comment (jacques-n) [reply binary/expression.proto:151]: I think that CaseValue (or SwitchValue) are an improvement over simply "switch". Just felt that ifthen was probably clearer to most people.

Comment (cpcloud) [reply binary/expression.proto]: Done

**Review by jacques-n — APPROVED** [review]

**Pull request merged**